src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -836,6 +840,10 @@ void network::build_exec_order() {
                     // i.e., We need to make sure that all the preds of them are already updated too.
                     for (auto dep : node->get_dependencies()) {
                         if (!dep.first->is_type<data>()) {
+                            if (dep.first->is_type<reshape>() && dep.first->can_be_optimized()) {
+                                auto reshape_node = dep.first;
+                                add_to_exec_order(reshape_node->get_dependency(0).id());
```

Comment:
Not adding reshape itself?

Reply:
It's not reshape itself, it is the input of reshape node.